msgpack-objc/MessagePack.m

```diff
@@ -9,17 +9,25 @@
 #import "MessagePack.h"
 #import <msgpack-c/msgpack.h>
 
-@interface TimestampExtension : MessagePackExtension
-
-- (instancetype)initWithDate:(NSDate *)date;
-- (NSDate *)date;
+@interface NSDate (TimestampSerializable) <MessagePackSerializable>
 
 @end
 
 @implementation MessagePack
 
+static NSDictionary<NSNumber *, Class> *_extensions;
```

Comment:
You could make this a class property! `@property (nonatomic, strong, class) NSDictionary<NSNumber *, Class> *extensions`.

Reply:
I was thinking about that too, but it will just be more code since you will have to implement the setter and getter and read/write to that static variable anyway. Do you see any benefits?

Reply:
Yeah it might be better like it is now. One thing to note is that it's currently not thread safe, would require a synchronize block inside the registerClass-function